Adds Backstage integration documentation (#1339)
DataDog/integrations-extras: backstage/README.md

# Agent Check: backstage

## Overview

[Backstage][1] is an open platform for building Developer Portals. This integration enables you to embed Datadog graphs and dashboards into your Backstage instance.

## Setup

### Installation

1. Install the Datadog plugin into Backstage:

```shell
cd packages/app
yarn add @roadiehq/backstage-plugin-datadog
```

2. Add the Datadog plugin widget to your Backstage Overview tab. See the [detailed instructions][2] for more information.
3. Find or create the [public URL][3] for your Datadog dashboard. 
4. Add the dashboard URL to the plugin's metadata:

```yaml
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: sample-service
  description: |
    A sample service
  annotations:
    datadoghq.com/dashboard-url: <DATADOGURL>
```

> ### Verifiaction

Reviewer: ```suggestion
### Verification
```

> Open your Backstage instance Overview tab and see if your Datadog dahsboard or graph is rendered as expected.

Reviewer: ```suggestion
Open your Backstage instance Overview tab to confirm your Datadog dashboard or graph is rendered as expected.
```

# Agent Check: backstage

## Overview

[Backstage][1] is an open platform for building Developer Portals. This integration enables you to embed DataDog graphs and dashboards into your Backstage instance.

## Setup

### Installation

1. Install the DataDog plugin into Backstage:

```shell
cd packages/app
yarn add @roadiehq/backstage-plugin-datadog
```

2. Add the DataDog plugin widget to your Backstage Overview tab ([detailed instructions][2]).
3. Get the [public URL][3] for your DataDog Dashboard. 
4. Add the Dashboard URL to the plugin's metadata:

```yaml
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: sample-service
  description: |
    A sample service
  annotations:
    datadoghq.com/dashboard-url: <<DATADOGURL>>
```

### Screenshot

> ![Screenshot of Backstage's Overview featuring a DataDog graph](https://raw.githubusercontent.com/RoadieHQ/roadie-backstage-plugins/main/plugins/frontend/backstage-plugin-datadog/docs/datadog-widget.png)

Reviewer 2: Can you move this image into our repo so it can be displayed correctly? Thanks!

Author: I'll remove the image as suggested by @buraizu, thanks!

> ![Screenshot of Backstage's Overview featuring a DataDog graph](https://raw.githubusercontent.com/RoadieHQ/roadie-backstage-plugins/main/plugins/frontend/backstage-plugin-datadog/docs/datadog-widget.png)

Reviewer: Would you be able to provide a more descriptive title and additional detail to this section? Currently it's not clear what information is meant to be conveyed by the screenshot. 

For example, would a customer see something similar to the screenshot after successfully setting this up? If so, please update the title of this section to `Validation` or something similar, as in this other example from the `integrations-extras` repository:
https://github.com/DataDog/integrations-extras/tree/master/kernelcare#validation

With that being said, we generally caution against including screenshots in documentation, as these are difficult to maintain due to UI changes. Please consider if the content of this screenshot can be adequately expressed in a written form, and substitute the written description for the screenshot if so.
